src/clj/api/mutations.clj

```diff
@@ -88,7 +89,8 @@
       (clj->gql (continuous-tree-model/get-tree db {:id id}))
       (catch Exception e
         (log/error "Exception occured" {:error e})
-        (errors/handle-analysis-error! db id e)))))
+        (errors/handle-analysis-error! db id e)
+        (throw e)))))
```

Comment:
maybe we could refactor the try-catch to also be a decorator?
```
(try
  (resolver-fn context args values)
    (catch Exception e
      (log/error "Exception occured" {:error e})
      (errors/handle-analysis-error! db id e)
      (throw e)))
```

then the composed decorator becomes `(comp auth-decorator try-catch-decorator tx-decorator)`, and should both handle the error (persist it) and rollback the initial tx, WYT?

Reply:
I don't think this is a good idea since you need to remember to add the decorator to every mutation that deals with analysis processing, and doesn't solve the general "don't forget to bubble up exceptions" problem